### PR TITLE
gloo-ctl 0.8.3 (new formula)

### PR DESCRIPTION
**Pull request by scranton**
This is the command line tool for the [Gloo](https://github.com/solo-io/gloo) OSS project. This command line helps install and manage Gloo on Kubernetes based platforms.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/gloo-ctl.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    cd dir do
+      # Make binary
+      system "dep", "ensure", "-vendor-only"
+      system "make", "glooctl", "TAGGED_VERSION=v0.8.3", "RELEASE=false"
+      bin.install "_output/glooctl"
+    end
+  end
+
+  test do
+    run_output = shell_output("#{bin}/glooctl 2>&1")
```
Comment (fxcoudert) [inline Formula/gloo-ctl.rb]: We need a test that exercises the some of the functionality of the app.

File: Formula/gloo-ctl.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/solo-io/gloo"
+    dir.install buildpath.children - [buildpath/".brew_home"]
+
+    cd dir do
+      # Make binary
+      system "dep", "ensure", "-vendor-only"
+      system "make", "glooctl", "TAGGED_VERSION=v0.8.3", "RELEASE=false"
```
Comment (fxcoudert) [inline Formula/gloo-ctl.rb]:
```suggestion
      system "make", "glooctl", "TAGGED_VERSION=v#{version}", "RELEASE=false"
```

Comment (fxcoudert) [reply Formula/gloo-ctl.rb]: Why `RELEASE=false`?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/gloo-ctl.rb (hunk 2 above)
Comment (scranton) [reply Formula/gloo-ctl.rb]: RELEASE flag is tied to our internal release system. we need to think about refactoring makefile.

File: Formula/gloo-ctl.rb (hunk 1 above)
Comment (scranton) [reply Formula/gloo-ctl.rb]: the `glooctl` tool depends on access to a Kubernetes cluster to do anything, so hard to make assumptions. so other than `glooctl help` and `glooctl --version`, there isn't functionality to test without access to a kubernetes cluster.

This is very similar to kubernetes-cli.eb (kubectl), and my tests are modeled after that script.

Comment (fxcoudert) [reply Formula/gloo-ctl.rb]: Try to perform some operation and check that it fails, then. `kubect` is not a good example, we know its test is bad.

Comment (scranton) [reply Formula/gloo-ctl.rb]: I've added another test as you asked and validated it fails as expected.

File: Formula/gloo-ctl.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
```
Comment (fxcoudert) [inline Formula/gloo-ctl.rb]: Use `#{version}` instead of hardcoding version here.

File: Formula/gloo-ctl.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      bin.install "_output/glooctl"
+    end
+  end
+
+  test do
+    run_output = shell_output("#{bin}/glooctl 2>&1")
+    assert_match "glooctl is the unified CLI for Gloo.", run_output
+
+    version_output = shell_output("#{bin}/glooctl --version 2>&1")
+    assert_match "glooctl version 0.8.3", version_output
```
Comment (fxcoudert) [inline Formula/gloo-ctl.rb]: Use `#{version}` instead of hardcoding version here.

File: Formula/gloo-ctl.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/solo-io/gloo"
+    dir.install buildpath.children - [buildpath/".brew_home"]
+
+    cd dir do
+      # Make binary
```
Comment (fxcoudert) [inline Formula/gloo-ctl.rb]: Remove that comment

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Merging! Thanks @scranton for your contribution to Homebrew, we look forward to see you again in future pull requests :)

**Comment by scranton** [comment]
@fxcoudert Thanks for walking me through my first formula submit!

**Pull request closed without merging**